### PR TITLE
fix(release): integrate v0.4.1 stabilization fixes

### LINKED ISSUE #379: bug(db): v0.4.0 rejects databases evolved through released v0.3.26 migrations
## Why

ProjectAtlas v0.4.0 rejects a valid schema-8 database evolved through released v0.3 versions because compatibility preflight compares one historical physical column order with a canonical order before migration can run. This blocks safe upgrade from the supported v0.3.26 compatibility floor and puts authored local state behind a rebuild workaround.

## What Changes

- Admit every schema-8 physical layout that released v0.3.26 accepts when its named schema contract is semantically equivalent.
- Migrate admitted databases atomically through the existing schema-8-to-16 sequence.
- Keep automatic writable-open detection generic for every version in the centralized migration inventory; schema 8 only repairs admission of its released physical layouts.
- Preserve approved purposes, health resolutions, telemetry, project identity, root binding, and durable source rows.
- Invalidate incompatible predecessor publication markers after successful migration so the next scan publishes the schema-16 contract; preserve those markers unchanged on rollback.
- Continue rejecting unknown, corrupt, missing, extra, renamed, or semantically incompatible schema objects without resetting the database.

## Capabilities

- Released SQLite predecessor compatibility
- Transactional schema migration and rollback
- Authored-state, telemetry, identity, and source-row preservation
- Project-local `init` and `scan` upgrade behavior

## Architecture Diagrams

- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)

## Release Scope

Bugfix-only v0.4.1 work in `projectatlas-db` schema preflight, released schema-8 fixtures, and public `init`/`scan` compatibility tests. No new dependency, table, index, schema version, command, or migration framework.

## Non-Goals

- Accepting arbitrary schema drift, corruption, or unsupported versions.
- Resetting, replacing, or silently rebuilding an incompatible database.
- Reconstructing authored purposes, identity, health resolutions, or telemetry from derived source state.
- Retaining stale schema-8 derived publication trust under the schema-16 contract.
- Adding a second migration path outside `projectatlas-db`.

## Pre-Mortem

Likely failure modes:

- A fix handles only one copied SQL string and misses another released compatible column order.
- Migration preserves rows but changes defaults, indexes, identity, or publication semantics.
- A late failure exposes a partial schema or generation.
- The database path works in storage tests but fails through `init` or `scan`.

Mitigations:

- [x] Compare complete named semantics across both released layouts and reject incompatible drift. (OpenSpec tasks: 1.2, 2.1)
- [x] Verify durable rows, defaults, indexes, identity, publication invalidation/republication, integrity, reopen, and root ownership. (OpenSpec tasks: 2.2, 3.1)
- [x] Inject late failure and prove transaction rollback plus independent project isolation. (OpenSpec tasks: 2.3)
- [x] Exercise the public `init`/`scan` boundaries and full release-quality gates. (OpenSpec tasks: 3.1, 3.2)

## OpenSpec Tasks

OpenSpec change: `migrate-released-database-layouts`

### 1. Compatibility Fixtures and Routing

- [x] 1.1 Map `migrate-released-database-layouts` to GitHub issue #379 in `openspec/issue-map.json` and keep the issue checklist synchronized with this file.
- [x] 1.2 Add the captured released v0.3.11-to-v0.3.26 schema-8 layout plus focused positive and incompatible-drift tests that reproduce the physical-order rejection before the production fix.

### 2. Semantic Admission and Atomic Migration

- [x] 2.1 Compare schema-8 predecessor columns and required index references by complete named semantic contract while retaining exact validation for current/later schemas and strict rejection of missing, extra, renamed, or changed objects.
- [x] 2.2 Prove successful migration preserves approved purposes, health resolutions, telemetry, project identity, root binding, and durable source rows through schema 16; invalidates predecessor publication trust; accepts a new complete generation; and passes integrity checks, reopen, and `root verify`.
- [x] 2.3 Inject a post-admission migration failure and prove rollback leaves the original schema, durable rows, publication state, root identity, and independent project databases unchanged.

### 3. Public Boundaries and Verification

- [x] 3.1 Cover read-only upgrade-required/no-write behavior and real CLI `init`/`scan` migration behavior for both released schema-8 layouts.
- [x] 3.2 Run focused database/CLI tests, `cargo fmt --check`, workspace check/clippy/test/doc gates, OpenSpec validation, IssueOps, and live review-feedback reconciliation.

### LINKED ISSUE #380: bug(analysis): impact dead-code analysis ignores deadline and blocks for 300 seconds
## Why

Impact analysis with dead-code candidates can ignore a one-second request deadline until the MCP host reaches its five-minute timeout. ProjectAtlas must own the complete bounded operation and return typed deadline or cancellation state promptly instead of relying on the host to abort it.

## What Changes

- Apply one captured absolute deadline, cancellation source, and aggregate resource budget across candidate discovery, SQLite reads, traversal, hydration, composition, and rendering.
- Stop database iteration and service work promptly with typed deadline, cancellation, truncation, coverage, confidence, and total-state results.
- Release SQLite statements and snapshots, task records, and intermediate allocations before returning.
- Preserve equivalent behavior through CLI and MCP, including immediate follow-up responsiveness.

## Capabilities

- Bounded graph impact and dead-code analysis
- SQLite deadline and cancellation propagation
- Typed partial and truncated analysis results
- CLI and MCP adapter equivalence and cleanup

## Architecture Diagrams

- [Agent navigation views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/agent-navigation.md#the-progressive-navigation-sieve)
- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)

## Release Scope

Bugfix-only v0.4.1 work in the existing service analysis path and owning bounded database reads. No new tool, request field, executor, task registry, schema, index, or background work.

## Non-Goals

- Removing impact or dead-code analysis.
- Raising MCP host timeouts.
- Returning unbounded or falsely exact partial graphs.
- Weakening freshness, confidence, snapshot consistency, or source authority.

## Pre-Mortem

Likely failure modes:

- Deadline checks stop traversal but leave discovery or hydration unbounded.
- Early return retains SQLite or task state.
- Timing coverage becomes flaky or misses a phase.
- CLI and MCP adapters diverge.

Mitigations:

- [x] Expire deterministic tests in discovery, SQLite work, traversal, hydration, composition, and rendering while threading one control. (OpenSpec tasks: 1.2, 2.1, 2.2)
- [x] Assert snapshot, statement, task, and allocation cleanup plus immediate follow-up responsiveness. (OpenSpec tasks: 2.2, 3.1)
- [x] Combine stage hooks with tolerant real-adapter elapsed assertions. (OpenSpec tasks: 1.2, 3.1)
- [x] Preserve one typed service result through equivalent CLI and MCP coverage and full gates. (OpenSpec tasks: 2.3, 3.1, 3.2)

## OpenSpec Tasks

OpenSpec change: `bound-impact-analysis-deadlines`

### 1. Reproduction and Ownership
- [x] 1.1 Map `bound-impact-analysis-deadlines` to GitHub issue #380 in `openspec/issue-map.json` and keep the issue checklist synchronized with this file.
- [x] 1.2 Add deterministic leaf and larger-entrypoint regressions that expire during dead-code discovery, SQLite work, traversal, hydration, and output, plus a successful non-expired control.

### 2. Complete Aggregate Control
- [x] 2.1 Thread the existing captured request control through every impact/dead-code candidate, traversal, hydration, composition, and rendering phase using one absolute deadline and cancellation source.
- [x] 2.2 Make every owning SQLite read bounded and interruptible at its batch/iteration boundary, and prove statements, snapshots, task records, and intermediate collections are released on terminal control.
- [x] 2.3 Preserve typed deadline, cancellation, truncation, coverage, confidence, total-state, and safe-partial semantics through the shared service result and equivalent CLI/MCP adapters.

### 3. Adapter and Release Verification
- [x] 3.1 Add real MCP and CLI elapsed-tolerance tests, immediate follow-up responsiveness, no partial authoritative mutation, and declared row/node/edge/visited/intermediate/output budget coverage.
- [x] 3.2 Run focused service/database/CLI/MCP tests, `cargo fmt --check`, workspace check/clippy/test/doc gates, OpenSpec validation, IssueOps, and live review-feedback reconciliation.

### LINKED ISSUE #382: bug(scan): recursively index nested folders without crossing worktree boundaries
## Why

ProjectAtlas must recursively index eligible nested source while preventing alternate branch checkouts from contaminating one atlas. The current repository is protected by `/.worktrees/` in `.gitignore`, but ignore policy alone is not a sufficient product guarantee: an unignored linked worktree beneath a selected root can otherwise look like ordinary nested source.

## What Changes

- Preserve recursive traversal, Git/ProjectAtlas ignore pruning, symlink containment, and per-root database identity. - Discover linked worktrees registered to the same common Git repository once before traversal and prune every other registered worktree beneath the selected root, independent of folder name or ignore policy. - Leave unrelated nested repositories and submodules compatible. - Return typed non-mutating `init_required` guidance for an uninitialized selected worktree and make the shipped agent workflow invoke `atlas_init` automatically for that exact `project_path`. - Prove worktree-local init, scan, watch, MCP routing, branch bytes, database/config/identity isolation, bare-root guidance, and repository ignore defense in depth.

## Capabilities

- Recursive repository scanning - Structural linked-worktree isolation - Automatic per-worktree initialization - Git and ProjectAtlas ignore enforcement - Per-project database, purpose, source, summary, symbol, and graph identity - Cross-platform CLI/MCP worktree routing

## Architecture Diagrams

- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)
- [Agent navigation views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/agent-navigation.md#the-progressive-navigation-sieve)

## Release Scope

Essential v0.4.1 correctness work in bounded Git policy discovery, existing filesystem excluded prefixes, root/missing-atlas guidance, the shipped skill, and real Git integration coverage. Existing scanner and database boundaries remain the owners.

## Non-Goals

- Enumerating or indexing every linked worktree from one command. - Sharing one database across branches or storing it in the common Git directory. - Hard-coding `.worktrees` or another personal directory name into reusable scanner logic. - Excluding unrelated nested repositories merely because they contain Git metadata. - Discovering a repository root from an arbitrary descendant. - Silently creating project state from an ordinary read-only command.

## Pre-Mortem

Likely failure modes:

- An unignored registered worktree is still traversed as ordinary source.
- Structural pruning also excludes an unrelated nested repository or the selected worktree itself.
- Missing-atlas guidance still leaves humans choosing database filenames or writes during a read.
- CLI and MCP select different roots or one process-global root leaks between worktrees.
- Repository ignore defense in depth is removed unnoticed.

Mitigations:

- [x] Reproduce with a real unignored in-root worktree and derive same-repository registered roots before traversal. (OpenSpec tasks: 1.2, 2.1, 3.1)
- [x] Cover selected-root retention, unrelated nested Git compatibility, and typed boundary uncertainty. (OpenSpec tasks: 2.1, 3.1)
- [x] Return exact non-mutating init guidance and make the shipped skill invoke worktree-local init automatically. (OpenSpec tasks: 2.2, 2.3)
- [x] Interleave CLI/MCP init, scan, watch, and per-call routing across two identities. (OpenSpec tasks: 2.3, 3.1)
- [x] Retain and test this repository's versioned ignore policy as a second layer. (OpenSpec tasks: 3.1, 3.2)

## OpenSpec Tasks

OpenSpec change: `isolate-linked-worktree-atlases`

### 1. Real Worktree Reproduction and Routing
- [x] 1.1 Map `isolate-linked-worktree-atlases` to GitHub issue #382 in `openspec/issue-map.json` and keep the issue checklist synchronized with this file.
- [x] 1.2 Add a real `git worktree add` integration fixture without a worktree-container ignore rule and prove the main checkout currently risks admitting branch-only sibling source before the structural exclusion fix.

### 2. Structural Isolation and Automatic First Use
- [x] 2.1 Reuse bounded common-Git/worktree policy discovery to derive same-repository registered worktree roots once, prune every other in-root worktree through existing excluded prefixes, fail typed on boundary uncertainty, and leave unrelated nested repositories/submodules compatible.
- [x] 2.2 Return typed non-mutating `init_required` guidance for read-only use of an uninitialized selected worktree and make the shipped agent workflow automatically invoke `atlas_init` for that exact `project_path`.
- [x] 2.3 Prove CLI `init`, first-write `scan`, `watch --once`, and interleaved MCP per-call `project_path` retain independent worktree-local DB/config/host-cwd/identity/purpose/source/summary/symbol/graph state and typed bare-root guidance.

### 3. Recursive Coverage and Verification
- [x] 3.1 Cover deep eligible folders, ignore-before-descent, repository `/.worktrees/` defense-in-depth policy, branch-only clean/dirty/add/edit/delete/switch behavior, sibling/common-Git exclusion without ignore, and unrelated nested-Git compatibility on Windows and Linux.
- [x] 3.2 Run focused filesystem/CLI/MCP/E2E tests, `cargo fmt --check`, workspace check/clippy/test/doc gates, OpenSpec validation, IssueOps, and live review-feedback reconciliation.

### LINKED ISSUE #383: bug: Windows installer validates child PATH while Codex parent remains stale
## Why

When Windows locks the stale LocalAppData stable mirror, the installer safely uses the versioned runtime and absolute MCP paths but can validate bare `projectatlas` only inside its own mutated child process. The already-running Codex parent can still launch later sibling shells with stale PATH, and a restart is only a valid remedy when the installer persisted a current runtime for future processes.

## What Changes

- Capture the parent PATH and bare-command identity before mutation, then resolve the effective sibling command after installer-owned stale-shim quarantine without executing it.
- Distinguish verified runtime plus generated-config readiness, installer-process CLI readiness, unchanged parent CLI readiness, and whether restart is an effective repair; keep optional global Codex registry repair separately reported.
- Derive parent state from what a later sibling on the captured PATH will actually resolve: the verified versioned runtime or a stable mirror synchronized during installation.
- Emit restart-required only when the reconstructed fresh Machine-plus-User PATH resolves the verified runtime. For supplied-runtime, managed-PATH, or Machine-shadowed installs where restart is insufficient, require the stale command to be unlocked or removed before rerunning.
- Preserve persisted User PATH where enabled, versioned runtime installation, and absolute Codex/Claude/OpenCode MCP configuration.
- Model the persistent parent, installer child, and later sibling process topology.

## Capabilities

- Windows plugin/runtime installation
- Stable-mirror synchronization
- Bare CLI runtime resolution
- Parent-host readiness and repair state
- Absolute generated and registered MCP integration

## Architecture Diagrams

- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)

## Release Scope

Bugfix-only v0.4.1 work in `plugins/projectatlas/scripts/install-runtime.ps1`, matching operator guidance, and existing Windows installer trust/E2E coverage. Rust runtime, SQLite, MCP request schemas, and generated config shapes remain unchanged.

## Non-Goals

- Mutating another process's environment.
- Terminating any pre-existing ProjectAtlas, Codex, terminal, or unrelated process. The installer may terminate only the exact runtime-info probe process tree it started after that probe exceeds its time or output bound.
- Treating a usable absolute-runtime/MCP installation as failed solely because the parent needs a restart or bare-command repair.
- Replacing absolute version-guarded MCP paths with bare command resolution.
- Adding a cross-platform installer result framework.

## Pre-Mortem

Likely failure modes:

- The installer still infers parent readiness from its child-only PATH.
- A synchronized mirror is treated as sufficient even when the parent does not resolve that path.
- Restart is promised when persistence was skipped or an earlier Machine PATH command still shadows the persisted User PATH runtime.
- A locked mirror becomes a hard failure despite safe runtime/MCP readiness.
- Absolute generated configs or Codex registration regress.
- The fix kills a pre-existing process, fails to bound its exact installer-owned probe process tree, or reports restart forever after a fresh host starts.
- Script behavior passes locally but is absent from Windows release smoke.

Mitigations:

- [x] Model the persistent parent, installer child, and later sibling while deriving state from the captured parent PATH after quarantine. (OpenSpec tasks: 1.2, 2.1)
- [x] Keep exit-zero runtime/generated-config readiness explicit and separate from parent readiness and actionable restart state. (OpenSpec tasks: 2.2, 3.1)
- [x] Reverify every generated and registered absolute host path in locked and unlocked cases. (OpenSpec tasks: 2.2, 3.1)
- [x] Assert fixture-process survival, fresh-host convergence, Machine-PATH shadow rejection, restart-insufficient supplied-runtime behavior, no pre-existing or unrelated process termination, and bounded owned-probe termination. (OpenSpec tasks: 3.1)
- [x] Run the focused Windows release-quality gates and review reconciliation. (OpenSpec tasks: 3.2)

## OpenSpec Tasks

OpenSpec change: `report-installer-host-restart`

### 1. Persistent-Host Reproduction

- [x] 1.1 Map `report-installer-host-restart` to GitHub issue #383 in `openspec/issue-map.json` and keep the issue checklist synchronized with this file.
- [x] 1.2 Extend the Windows locked-mirror regression to model one persistent parent environment, the installer child, and a later sibling bare-CLI probe, proving the child-only success claim before the fix.

### 2. Truthful Installer Readiness

- [x] 2.1 Capture inherited bare-command identity before PATH mutation, report stable-mirror synchronization from the existing helper, and derive restart-required state only when the unchanged parent would remain stale.
- [x] 2.2 Emit one deterministic final runtime/generated-config, installer-CLI, and host-restart readiness record plus clear required warning while preserving exit-zero partial success, persisted User PATH, and absolute Codex/Claude/OpenCode MCP paths.

### 3. Safety and Verification

- [x] 3.1 Cover unlocked mirror, already-current inherited PATH, fresh-host convergence, locked-process survival, no pre-existing or unrelated process termination, bounded owned-probe termination, and unchanged generated-config/registry verification.
- [x] 3.2 Run focused Windows installer trust/E2E tests, script syntax/static checks, affected Rust check/clippy/test/doc gates, OpenSpec validation, IssueOps, and live review-feedback reconciliation.

### LINKED ISSUE #385: bug(graph): configured module aliases leave semantic relations unresolved
## Why

ProjectAtlas v0.4.0 parses configured ECMAScript imports but cannot resolve `tsconfig.json` or `jsconfig.json` aliases. Real callers can therefore disappear from inbound, impact, and dead-code analysis even though the equivalent relative imports already resolve.

## What Changes

- Load bounded, validated `baseUrl` and `paths` mappings from the nearest applicable `tsconfig.json` or `jsconfig.json`.
- Supply one deterministic configuration snapshot to the shared ECMAScript resolver used by JavaScript, JSX, TypeScript, TSX, and embedded Vue source.
- Reuse existing extension, `index`, ambiguity, unresolved, file, and exact-symbol graph ownership.
- Treat compiler-configuration creation, edits, renames, and removal as complete derived-refresh inputs.
- Preserve Python, Rust, and Cargo provider contracts without applying ECMAScript rules to them.

## Capabilities

- Configured ECMAScript module resolution
- Alias-aware file and exact-symbol graph edges
- Alias-safe impact and dead-code analysis
- Atomic configuration refresh across CLI and MCP

## Architecture Diagrams

- [Agent navigation views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/agent-navigation.md#the-progressive-navigation-sieve)
- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)

## Release Scope

Bugfix-only v0.4.1 work in the existing symbol and CLI runtime boundaries. No new crate, dependency, database schema, migration, graph storage model, CLI command, or MCP tool.

## Non-Goals

- Runtime-only aliases absent from repository configuration.
- Arbitrary bundler-plugin or JavaScript configuration execution.
- A parser-local TypeScript or Vue special case.
- Semantic-support claims for detection-only or fallback-only languages.

## Pre-Mortem

Likely failure modes:

- Broad mappings invent resolution or ambiguity.
- Malformed or escaping configuration reaches publication.
- Incremental refresh retains stale callers after a config change.
- ECMAScript behavior is incorrectly claimed for other providers.

Mitigations:

- [x] Prevent broad or overlapping mappings from inventing resolution by selecting the nearest config and most-specific pattern while retaining ambiguity. (OpenSpec tasks: 1.1, 1.5)
- [x] Reject malformed, escaping, over-limit, or concurrently changed configuration before publication. (OpenSpec tasks: 1.2, 1.3, 1.5)
- [x] Force a complete derived refresh when compiler configuration changes so stale callers cannot survive incrementally. (OpenSpec tasks: 1.3, 1.5)
- [x] Keep Python, Rust, and Cargo semantics unchanged and explicitly scoped. (OpenSpec tasks: 1.4, 1.5)
- [x] Keep the complete language, adapter, failure, and compatibility surface green through final hosted review. (OpenSpec tasks: 1.5, 1.6)

## OpenSpec Tasks

OpenSpec change: `resolve-configured-module-aliases`

### 1. OpenSpec Task Checklist
- [x] 1.1 Add a typed, bounded configured-module projection context in `projectatlas-symbols`; resolve nearest applicable ECMAScript `baseUrl` and most-specific `paths` candidates through shared import and call keys while preserving relative, extension, `index`, unresolved, and ambiguous behavior.
- [x] 1.2 Add bounded JSONC `tsconfig.json` / `jsconfig.json` discovery and repository-contained normalization in the CLI runtime; share one deterministic snapshot across full and incremental graph projection without adding a crate, dependency, schema, or transaction owner.
- [x] 1.3 Treat compiler-configuration creation, edits, renames, and removal as complete-refresh inputs; reject malformed, escaping, over-limit, or concurrently changed configuration before atomic publication.
- [x] 1.4 Audit Python, Rust, and Cargo semantic providers, retain applicable compatibility coverage, and record ECMAScript compiler aliases as not applicable without broadening provider claims.
- [x] 1.5 Add coherent unit, integration, and real CLI/MCP fixtures covering JavaScript, JSX, TypeScript, TSX, Vue, file and exact-symbol traversal, outbound resolution, impact/dead-code safety, extension and `index` inference, nearest precedence, missing/no-config controls, ambiguity, malformed input, configuration edit/removal, bounds, cancellation, atomicity, and adapter equivalence.
- [x] 1.6 Pass focused crate and E2E tests, formatting, workspace check/clippy, strict OpenSpec validation, issue-checklist validation, and live GitHub review-feedback audit without running benchmarks; leave task completion and remote issue state for exact-head parent review.

### LINKED ISSUE #386: bug(codex): purpose-review item schema collapses to Array<unknown>
## Why

ProjectAtlas v0.4.0 emits standards-valid typed MCP input schemas, but the supported Codex tool-schema bridge does not reliably retain local `$defs` and `$ref` members. Nested inputs such as `atlas_purpose_review.items` can therefore appear as `Array<unknown>` even though runtime deserialization remains strict.

## What Changes

- Audit the complete advertised MCP input inventory and keep every current Codex-facing schema self-contained.
- Inline the three affected small nested types at their existing Schemars ownership boundary without changing Serde or runtime request types.
- Keep `atlas_purpose_review.items` concrete with required `path` and optional purpose and conditional-write fields.
- Add packaged raw-stdio coverage that rejects local schema references, verifies the purpose-review shape, and confirms `tools/list` creates no project index state.
- Preserve raw MCP rejection, conditional-write admission, wrong-root, missing-index, and no-implicit-mutation behavior.
- Treat general Codex support for standards-valid local references as an upstream host dependency; ProjectAtlas supplies only the narrow compatibility representation it owns.

## Capabilities

- MCP tool-schema compatibility
- Typed purpose-review requests
- Codex plugin integration
- Raw stdio contract validation

## Architecture Diagrams

- [ProjectAtlas architecture views](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#architecture-views)

## Release Scope

Bugfix-only v0.4.1 work in the existing CLI/MCP schema annotations and packaged MCP contract tests. No database schema, storage behavior, CLI contract, dependency, tool inventory, or broad MCP redesign is included.

## Non-Goals

- Replacing typed request structs with untyped JSON values.
- Building a generic JSON Schema resolver or rewriting framework.
- Weakening runtime validation in reliance on host-side validation.
- Folding the broader tool-surface redesign from #310 into this bug.
- Claiming that local ProjectAtlas tests fix or prove Codex's general schema renderer.

## Pre-Mortem

Likely failure modes:

- Only purpose review is repaired while another advertised tool retains a bridge-sensitive local reference.
- A generic schema walker duplicates Schemars and creates a second schema authority.
- The exposed item object loses required `path` or makes optional conditional-write fields required.
- Host-readable schemas mask weaker runtime admission or allow an implicit database mutation.
- Raw stdio shape evidence is overstated as proof of Codex's upstream renderer.
- The packaged tool inventory changes without updating its compatibility contract.

Mitigations:

- [x] Reproduce the raw boundary, audit the complete inventory, and inline the exact affected typed schemas. (OpenSpec tasks: 1.1, 1.2, 2.1)
- [x] Gate every advertised input and verify the concrete purpose-review required and optional fields. (OpenSpec tasks: 2.2, 3.1)
- [x] Preserve typed runtime rejection, conditional writes, root isolation, missing-index behavior, and no mutation. (OpenSpec tasks: 2.3, 3.1)
- [x] Run the proportionate local Rust, OpenSpec, issue-map, and diff-quality gates. (OpenSpec tasks: 3.2)
- [x] Synchronize the hosted checklist and reconcile packaged-Codex and live review behavior. (OpenSpec tasks: 3.3)

## OpenSpec Tasks

OpenSpec change: `expose-concrete-codex-tool-schemas`

### 1. Reproduction and Contract

- [x] 1.1 Reproduce the v0.4.0 raw-stdio local-reference schema and identify the Codex bridge compatibility boundary.
- [x] 1.2 Audit all advertised MCP inputs for local `$defs` and `$ref` use and record the complete affected set.

### 2. Concrete MCP Schemas

- [x] 2.1 Inline the affected schemas at their nested typed-schema boundary without changing Serde or runtime request types.
- [x] 2.2 Add packaged raw-stdio coverage that rejects any advertised local input reference and verifies concrete purpose-review required and optional fields.
- [x] 2.3 Preserve and verify runtime rejection, conditional-write admission, wrong-root, missing-index, and no-implicit-mutation behavior.

### 3. Verification

- [x] 3.1 Run focused MCP schema and purpose-review tests, including the missing-`path` negative case and compatibility inventory.
- [x] 3.2 Run formatting, workspace check, Clippy, rustdoc, OpenSpec, issue-map/checklist diagnostics, and diff-quality gates proportionate to the change.
- [x] 3.3 Synchronize the GitHub checklist and obtain final integrated, packaged-Codex/hosted, and live review proof.

### LINKED ISSUE #391: bug: separate healthy parser recovery tolerance from hostile stall budget
## Why

The Windows parser-supervisor adversarial gate applied its 500 ms hostile-stall allowance to the healthy restart used to prove cleanup. Normal process launch or containment admission could exceed that test-only budget on a loaded Windows host after the hostile peer had already been handled correctly.

## What Changes

- Give only the post-failure healthy recovery probe a platform-tolerant no-progress allowance.
- Preserve the short hostile-case budgets, exact typed failures, single launch attempt, and existing deadline for each attempt.
- Keep production parser deadlines, protocol, containment, cancellation, and cleanup unchanged.
- Require repeated Windows adversarial success plus ordinary cross-platform CI.

## Capabilities

- Optional parser process supervision.
- Failure cleanup and healthy restart verification.
- Release-gate reliability on Windows.

## Architecture Diagrams

- [Cancellation, failure, and retry ownership](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#cancellation-failure-and-watch-retry)
- [Quality gates](https://github.com/styler-ai/ProjectAtlas/blob/dev/docs/projectatlas-3-architecture.md#quality-gates)

## Release Scope

v0.4.1 test-harness correction limited to the adversarial parser-supervisor gate. No production timeout or protocol behavior changes.

## Non-Goals

- Do not relax production parser deadlines, cancellation, protocol validation, containment, resource limits, or cleanup.
- Do not lengthen hostile-stall cases.
- Do not add retries that could mask process-lifecycle failures.

## Pre-Mortem

Likely failure modes:

- A broad timeout increase weakens hostile-case assertions.
- A retry masks leaked state after the hostile process.
- Windows scheduling pressure still makes the healthy recovery gate unreliable.

Mitigations:

- [x] Override only the healthy recovery case while retaining one launch attempt and its existing attempt deadline. (OpenSpec tasks: 1.1)
- [x] Keep hostile bounds, typed failures, containment, and cleanup checks unchanged. (OpenSpec tasks: 1.2)
- [x] Require repeated Windows adversarial success and ordinary cross-platform gates. (OpenSpec tasks: 1.3, 1.4)

## OpenSpec Task Checklist

- [x] 1.1 Give only the post-failure healthy recovery probe a platform-tolerant no-progress allowance while retaining one launch attempt and its existing attempt deadline.
- [x] 1.2 Preserve hostile-case no-progress bounds, exact typed failures, containment checks, and cleanup requirements.
- [x] 1.3 Verify repeated adversarial-suite success on Windows and, before the post-stall healthy launch, probe its configured no-progress allowance at a synthetic 550 ms progress age so the suite deterministically rejects the hostile 500 ms bound without consuming any of the real two-second recovery window.
- [x] 1.4 Run `cargo test --locked -p projectatlas-cli --test parser_supervisor_adversarial --all-features`, `cargo test --workspace --all-targets --all-features --locked`, and `cargo test --workspace --doc --all-features --locked` with a 20-minute timeout per command, then run `gh pr checks --watch --fail-fast` with a 60-minute timeout against the ordinary `01-CI` Windows/Linux/macOS x64/macOS arm64 gates without changing production behavior.

### PR DESCRIPTION
**Pull request by styler-ai**
## Summary

- integrate the v0.4.1 database migration, bounded impact analysis, linked-worktree isolation, installer restart reporting, configured module aliases, concrete Codex MCP schemas, and parser recovery fixes
- close the cross-branch review gaps for exact configured alias targets and typed bare/common-Git root refusal
- keep automatic released-schema migration centralized in projectatlas-db and preserve worktree-local atlas ownership

## Why

The individual bug-fix branches share compatibility and release boundaries, while branch protection requires each PR to be current with dev. This integration head provides one exact candidate for cross-platform review and avoids redundant release-gate cycles.

## Validation

- full repository pre-push gate passed locally
- workspace format, check, all-feature Clippy, strict strings, cargo-deny, tests, doc tests, and rustdoc passed
- 100 active CLI E2Es passed, including released schema migration, linked worktrees, configured aliases, impact cancellation, installer, MCP, and parser coverage
- OpenSpec and IssueOps synchronization passed
- candidate ProjectAtlas self-scan and low-purpose lint completed
- long navigation benchmarks were not run; release policy keeps them manual-only

Closes #379
Closes #380
Closes #382
Closes #383
Closes #385
Closes #386
Closes #391

### CODE REVIEW

**Comment by styler-ai** [comment]
@codex review

File: crates/projectatlas-cli/src/runtime.rs [hunk 1]
```diff
@@ -1842,6 +1902,183 @@ pub(crate) fn canonical_project_root(root: &Path) -> Result<PathBuf, CliError> {
...
+        && root
+            .file_name()
+            .and_then(std::ffi::OsStr::to_str)
+            .is_some_and(|name| name.eq_ignore_ascii_case(".git"))
+    {
+        return Ok(true);
+    }
+
+    effective_git_config_bare_setting(&control_root, &config)
+        .map(|bare| bare.unwrap_or(control_root == root))
```
[the remark this answers is not in the document]
Comment (styler-ai) [reply crates/projectatlas-cli/src/runtime.rs]: Fixed at the shared root classifier: any structurally complete selected Git control root now returns typed worktree_required guidance regardless of directory basename or core.bare. The regression creates a real repository with git init --separate-git-dir and asserts init/root selection refuse it without creating ProjectAtlas state. Focused and full pre-push gates pass.

**Comment by styler-ai** [comment]
@codex review

**Comment by styler-ai** [comment]
@codex review

**Comment by styler-ai** [comment]
@codex review
